src/coreclr/tools/Common/Compiler/MethodExtensions.cs

```diff
@@ -82,25 +87,43 @@ public static string GetUnmanagedCallersOnlyExportName(this EcmaMethod This)
         /// since all of their collection interface methods are sealed and implemented on the System.Array and
         /// System.Array&lt;T&gt; base types, and therefore we can minimize the vtable sizes of all derived array types.
         /// </summary>
-        public static bool CanMethodBeInSealedVTable(this MethodDesc method)
+        public static bool CanMethodBeInSealedVTable(this MethodDesc method, NodeFactory factory)
         {
-            bool isInterfaceMethod = method.OwningType.IsInterface;
+            Debug.Assert(!method.OwningType.ContainsSignatureVariables(treatGenericParameterLikeSignatureVariable: true));
+
+            TypeDesc owningType = method.OwningType;
 
-            // Methods on interfaces never go into sealed vtable
-            // We would hit this code path for default implementations of interface methods (they are newslot+final).
-            // Interface types don't get physical slots, but they have logical slot numbers and that logic shouldn't
-            // attempt to place final+newslot methods differently.
-            if (method.IsFinal && method.IsNewSlot && !isInterfaceMethod)
+            // Interface types don't have physical slots so we never optimize to sealed slots
+            if (owningType.IsInterface)
+                return false;
+
+            // Implementations of static virtual methods go into the sealed vtable.
+            if (method.Signature.IsStatic)
                 return true;
 
-            // Implementations of static virtual method also go into the sealed vtable.
-            // Again, we don't let that happen for interface methods because the slot numbers are only logical,
-            // not physical.
-            if (method.Signature.IsStatic && !isInterfaceMethod)
+            // If the owning type is already considered sealed, there's little benefit in placing the slots
+            // in the sealed vtable: the sealed vtable has these properties:
```

Comment:
What are the downsides of keeping this simple and placing them in the sealed vtable? Are you trying to save the extra pointer to the sealed vtable?

Reply:
Yep, the extra pointer to the vtable. The old logic was just `method.IsFinal && method.IsNewSlot` so we didn't even look whether the type is sealed. Now we end up doing this for a lot more methods, often on valuetypes where it makes little sense.